interop/test_utils.go

```diff
@@ -676,7 +677,7 @@ func DoPickFirstUnary(tc testgrpc.TestServiceClient) {
 	}
 }
 
-func doOneSoakIteration(ctx context.Context, tc testgrpc.TestServiceClient, resetChannel bool, serverAddr string, dopts []grpc.DialOption) (latency time.Duration, err error) {
+func doOneSoakIteration(ctx context.Context, tc testgrpc.TestServiceClient, resetChannel bool, serverAddr string, dopts []grpc.DialOption) (latency time.Duration, peer peer.Peer, err error) {
```

Comment:
I admit this is existing art, but naked returns are not ideal for anything other than really small functions.
See: go/go-style/decisions#named-result-parameters

Not asking you change that in this PR, but if we can avoid adding more named result parameters to this function, that would be nice.

One option is to pass call options to this function. So, something like this:
```golang
func doOneSoakIteration(ctx context.Context, tc testgrpc.TestServiceClient, resetChannel bool, serverAddr string, dopts []grpc.DialOption, copts []grpc.CallOption) (latency time.Duration, err error) {
  ...
  reply, err = client.UnaryCall(ctx, req, copts...)
  ...
}
```

And then, the caller can create the `peer.Peer`, pass it as a `CallOption` to this function, and inspect the former after the RPC. What do you think?

Reply:
Good idea - done